src/nvim/eval.c

```diff
@@ -17203,6 +17279,12 @@ void set_vim_var_nr(int idx, long val)
   vimvars[idx].vv_nr = val;
 }
 
+/// Set special v: variable to "val"
+void set_vim_var_special(const int idx, const SpecialVarValue val)
```

Comment:
We could use `VimVarIndex` as type for `idx` for all the `*_vim_var*()` functions.